server/datastore/mysql/policies.go

```diff
@@ -697,6 +697,71 @@ func (ds *Datastore) IncrementPolicyViolationDays(ctx context.Context) error {
 	})
 }
 
+func (ds *Datastore) IncreasePolicyAutomationIteration(ctx context.Context, policyID uint) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		_, err := tx.ExecContext(ctx, `
+			INSERT INTO policy_automation_iterations (policy_id, iteration) VALUES (?,1)
+			ON DUPLICATE KEY UPDATE iteration = iteration + 1;
+		`, policyID)
+		return err
+	})
+}
+
+// OutdatedAutomationBatch returns a batch of hosts that had a failing policy.
+func (ds *Datastore) OutdatedAutomationBatch(ctx context.Context) ([]fleet.PolicyFailure, error) {
+	var failures []fleet.PolicyFailure
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		failures = failures[:0] // for retry
```

Comment:
Not sure I understand why we need to do this for retries, could you expand on it a bit?

Reply:
It's just if "withRetryTxx" retries the operation, we don't want the old data to stay inside.

Reply:
Oh right, it's a trim. Why not declare the var in this line?

Reply:
It's used in line 762. It's the output of the function.